File: C#/Katas/StringCalculator/StringCalculator.cs
namespace Katas
{
    public static class StringCalculator
    {
        public static int Add(string numbers, char seperator)
        {
            var result = 0;
            if (string.IsNullOrEmpty(numbers)) return result;

            foreach (var part in numbers.Split(seperator))
            {
                if (string.IsNullOrWhiteSpace(part)) continue;

                if (!int.TryParse(part, out var number)) continue;
                result += number;
            }

            return result;
        }
    }
}

File: Python/Katas/StringCalculator/stringcalculator.py
class stringcalculator(object):
    
    def add(numbers, delimeter):
        result = 0
        if numbers == "": return result
        
        for value in str(numbers).split(delimeter):
            if not stringcalculator.__check_int(value): continue
            result += int(value)

        return result 

    def __check_int(s):
        if s == "": 
            return False
        if s[0] in ('-', '+'):
    	    return s[1:].isdigit()
        return s.isdigit()



